htdocs/web_portal/controllers/downtime/add_downtime.php

```diff
@@ -71,53 +72,52 @@ function submit(\User $user = null) {
     if($confirmed == true){
         //Downtime is confirmed, submit it
         //$downtimeInfo = unserialize($_REQUEST['newValues']);  // didn't cater for UTF-8 chars
+        /**
+         * If confirmed by an user, submit the details of affected services
+         * and endpoints along with other details for each individual site.
+         */
         $downtimeInfo = json_decode($_POST['newValues'], TRUE);
         $serv = \Factory::getDowntimeService();
+        $downtimeInfo = unsetVariables($downtimeInfo, 'add');
+        $params = [];
+
+        foreach ($downtimeInfo['SITE_LEVEL_DETAILS'] as $siteID) {
+            $downtimeInfo['Impacted_Services'] = $siteID['services'];
+            $downtimeInfo['Impacted_Endpoints'] = $siteID['endpoints'];
+
+            $params['submittedDowntimes'][$siteID['siteName']] =
+                $serv->addDowntime($downtimeInfo, $user);
+        }
```

Comment:
Why the mix of capitalisation and underscores/camelCase? i.e. `SITE_LEVEL_DETAILS`, `Impacted_Services` and `submittedDowntimes`

Reply:
To keep consistency with user's web request ` getDtDataFromWeb()` in htdocs/web_portal/controllers/utils.php. I have used SITE_LEVEL_DETAILS.  That details would be coming from FORM Data.

And `Impacted_Services` is being used across codebase. So, I ignored the renaming.

Just like for any others params  I followed  normal naming convection